Tests/Microsoft.AppCenter.Crashes.Test.Windows/HandledErrorTest.cs

```diff
@@ -0,0 +1,237 @@
+            var storage = new Storage.Storage(new StorageAdapter("test.db"));
+            var ingestion = new IngestionHttp(mockHttpNetworkAdapter);
+            var channelGroup = new ChannelGroup(ingestion, storage, "app secret");
+            Crashes.Instance = new Crashes();
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(channelGroup, "app secret");
+            return mockHttpNetworkAdapter;
+        }
+
+        [TestMethod]
+        public void TrackErrorWithValidParameters()
+        {
+            var mockHttpNetworkAdapter = StartCrashes();
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<IDictionary<string, string>>(),
+                It.IsAny<string>(),
+                It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    actualLog = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs.Single() as HandledErrorLog;
+                    semaphore.Release();
+                });
+            var exception = new System.Exception("Something went wrong.");
+            var properties = new Dictionary<string, string> { { "k1", "v1" }, { "p2", "v2" } };
+            Crashes.TrackError(exception, properties);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait(1000);
+            Assert.IsNotNull(actualLog);
+            Assert.AreEqual(exception.Message, actualLog.Exception.Message);
+            CollectionAssert.AreEquivalent(properties, actualLog.Properties as Dictionary<string, string>);
+        }
+
+        [TestMethod]
+        public void TrackErrorWithTooManyProperties()
+        {
+            var mockHttpNetworkAdapter = StartCrashes();
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<IDictionary<string, string>>(),
+                It.IsAny<string>(),
+                It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    actualLog = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs.Single() as HandledErrorLog;
+                    semaphore.Release();
+                });
+            var exception = new System.Exception("Something went wrong.");
+            var maxProperties = 20;
```

Comment:
I'm not a fan of this pattern because it requires us to modify the tests if we change the number of max properties.  The test validation doesn't change though, so it's just making more work for us later by not using the constants.

I would prefer to make the constants `internal` instead of re-defining them

Reply:
There are pros and cons but I personally also don't like this one. Not a big deal though.

Reply:
I think that if we are relying on the constants we have defined in the code, then the test is quite circular. So I actually prefer redefinition in tests.

Reply:
Since this is more of an opinion type thing, I'm going to resolve the comments

Reply:
This is just less maintainable than making the constants internal, though.  I understand that by a "strict" definition it's circular, but I'm not sure that's a good reason to use this pattern.

If someone looks at this code and sees the number 20, they'll wonder where it's coming from.  There's no logical link between this value and the one in the test file, you just have to look in the file being tested.  What if the values aren't even in that file?  It becomes very difficult to link the verification value to the "source of truth"

Reply:
A test should not use constants from the SDK, it's a contract.
We already decided in GDPR test runner to not use the models, the same reasoning applies here.

Reply:
Found a compromise: We use the constants in tests for maintainability, but in order to make sure they still have the right values, we add specific tests for that elsewhere. Thus, we reduce the magic values to only one test, but we are still testing that they are correct.

Reply:
More context: https://stackoverflow.com/questions/3360074/do-you-use-constants-from-the-implementation-in-your-test-cases/23374335#23374335